tests/Core/Tokenizer/EnumTest.php

```diff
@@ -0,0 +1,163 @@
+
+        $this->assertArrayHasKey('scope_condition', $tokens[$enum]);
+        $this->assertArrayHasKey('scope_opener', $tokens[$enum]);
+        $this->assertArrayHasKey('scope_closer', $tokens[$enum]);
+
+        $this->assertSame($enum, $tokens[$enum]['scope_condition']);
+
+        $scopeOpener = $tokens[$enum]['scope_opener'];
+        $scopeCloser = $tokens[$enum]['scope_closer'];
+
+        $this->assertSame($expectedScopeOpenerLine, $tokens[$scopeOpener]['line']);
+        $this->assertArrayHasKey('scope_condition', $tokens[$scopeOpener]);
+        $this->assertArrayHasKey('scope_opener', $tokens[$scopeOpener]);
+        $this->assertArrayHasKey('scope_closer', $tokens[$scopeOpener]);
+        $this->assertSame($enum, $tokens[$scopeOpener]['scope_condition']);
+        $this->assertSame($scopeOpener, $tokens[$scopeOpener]['scope_opener']);
+        $this->assertSame($scopeCloser, $tokens[$scopeOpener]['scope_closer']);
+
+        $this->assertSame($expectedScopeCloserLine, $tokens[$scopeCloser]['line']);
+        $this->assertArrayHasKey('scope_condition', $tokens[$scopeCloser]);
+        $this->assertArrayHasKey('scope_opener', $tokens[$scopeCloser]);
+        $this->assertArrayHasKey('scope_closer', $tokens[$scopeCloser]);
+        $this->assertSame($enum, $tokens[$scopeCloser]['scope_condition']);
+        $this->assertSame($scopeOpener, $tokens[$scopeCloser]['scope_opener']);
+        $this->assertSame($scopeCloser, $tokens[$scopeCloser]['scope_closer']);
+
+    }//end testEnums()
+
+
+    /**
+     * Data provider.
+     *
+     * @see testEnums()
+     *
+     * @return array
+     */
+    public function dataEnums()
+    {
+        return [
+            [
+                '/* testPureEnum */',
+                5,
+                7,
+            ],
+            [
+                '/* testBackedIntEnum */',
+                10,
+                13,
+            ],
+            [
+                '/* testBackedStringEnum */',
+                17,
+                20,
+            ],
+            [
+                '/* testComplexEnum */',
+                24,
+                36,
+            ],
+            [
+                '/* testEnumWithEnumAsClassName */',
+                39,
+                39,
+            ],
+            [
+                '/* testDeclarationContainingComment */',
+                62,
+                64,
+            ],
+        ];
+
+    }//end dataEnums()
+
+
+    /**
+     * Test that "enum" when not used as the keyword is still tokenized as `T_STRING`.
+     *
+     * @param string $testMarker The comment which prefaces the target token in the test file.
+     *
+     * @dataProvider dataNotEnums
+     * @covers       PHP_CodeSniffer\Tokenizers\PHP::tokenize
+     *
+     * @return void
+     */
+    public function testNotEnums($testMarker)
+    {
+        $tokens = self::$phpcsFile->getTokens();
+
+        $target = $this->getTargetToken($testMarker, [T_ENUM, T_STRING]);
```

Comment:
```suggestion
        $target = $this->getTargetToken($testMarker, [T_ENUM, T_STRING], 'enum');
```

The test as-is may currently "find" a `T_STRING` token _before_ the `enum` target, which could lead to the wrong token being tested.

Also means that the data provider may need to provide an optional second parameter to provide the keyword in the case as used in the test for those tests not using lowercase.

Example - in this test:
```php
namespace My\Enum\Collection;
```

The test as it currently is, is testing `My`, not `Enum`.

Reply:
Fixed.

Reply:
By hard-coding the `enum` in the `testEnums()` test, no test can be added to ensure that the `enum` keyword not lowercase, is handled correctly.
This should be adjusted in a similar way as done for `testNotEnums()`, though it's perfectly fine to make the `$content` parameter for the test function optional with `'enum'` being the default value (for both test methods).

Reply:
Added case insensitive test.